source/generation_utils.py

```diff
@@ -0,0 +1,27 @@
+import os
+
+
+def define_generation_dir(model_repo_path):
+    #### to remove later ####
+    if model_repo_path == "models/model_2048_fake_wholedataset":
+        model_repo_path = "misnaej/the-jam-machine"
+    #### to remove later ####
+    generated_sequence_files_path = f"midi/generated/{model_repo_path}"
+    if not os.path.exists(generated_sequence_files_path):
+        os.makedirs(generated_sequence_files_path)
+    return generated_sequence_files_path
+
+
+def bar_count_check(sequence, n_bars):
+    """check if the sequence contains the right number of bars"""
+    sequence = sequence.split(" ")
+    # find occurences of "BAR_START" in a str
+    bar_count = 0
+    for seq in sequence:
+        if seq == "BAR_END":
```

Comment:
Why count bar end rather than bar start?

Reply:
> # find occurences of "BAR_END" in a "sequence"
    # I don't check for "BAR_START" because it is not always included in "sequence"
    # in some case BAR_START needs to be included the prompt (e.g. when generating one more bar)